### DISCUSSION TITLE
apache: Is it possible to include the "logs" directory in "persist"?

### DISCUSSION THREAD

**Discussion by hassy-6thsense** [ScoopInstaller/Scoop#5359] [Help and Questions]
When I install apache(httpd) with scoop, it creates "logs" directory inside "apps\apache\2.4.xx".
This "logs" is not a symbolic link but an entity, so if you upgrade apache with scoop, the log files before and after the update will be separated or disappear.
Is it possible to configure "logs" to be included in persist just like "conf" and "htdocs"?
I know that it can be achieved by specifying the absolute path in "httpd.conf", but if it can be included in persist in the scoop setting, I would like to adopt that method.

**Comment by rashil2000** [comment]
Yes, you can create an empty folder in the pre_install step, and then add it to the persist array.

**Comment by hassy-6thsense** [comment]
@rashil2000 
Thank you very much for your answer.
The way I do my own "pre_install" step and "persist" is by editing the JSON in the manifest.
But every time the upstream manifest is updated, the unique description disappears, so do I need to manually change the JSON file?
Or is there a mechanism to save the original description in another place and automatically merge it when "scoop update" is executed?
What I'm most happy about is adding "logs" to "persist" in apache.json in the main bucket, is that difficult?

Sorry for the elementary question.

**Comment by rashil2000** [comment]
> What I'm most happy about is adding "logs" to "persist" in apache.json in the main bucket, is that difficult?

No, just edit this file here https://github.com/ScoopInstaller/Main/blob/master/bucket/apache.json and create a pull request. It's very easy.

**Reply by hassy-6thsense** [reply]
So was it. I will create a PR, thank you very much!
